panel/src/components/Views/Preview/PreviewBrowser.vue

```diff
@@ -56,8 +55,25 @@ export default {
 		this.$events.off("content.publish", this.reload);
 	},
 	methods: {
+		onLoad() {
+			for (const link of this.window.document.querySelectorAll("a")) {
+				if (!link.href || link.onclick) {
+					continue;
+				}
+
+				if (link.href.startsWith(location.origin) === false) {
+					link.target = "_blank";
+					continue;
+				}
+
+				link.addEventListener("click", (e) => {
+					e.preventDefault();
+					this.$emit("navigate", link.href);
+				});
```

Comment:
I know that you had issues with event delegation, but I just realized that this will only work for elements that are present on load. For links that get injected after ajax calls or other more reactive JS parts, this will not be working.

Reply:
@bastianallgeier Maybe our best chance is to act on both? Pushed a version.